contracts/BIFI/strategies/Sonne/StrategyLendingSonne.sol

```diff
@@ -0,0 +1,407 @@
+        }
+        updateBalance();
+    }
+
+    function harvest() external virtual {
+        _harvest(tx.origin);
+    }
+
+    function harvest(address callFeeRecipient) external virtual {
+        _harvest(callFeeRecipient);
+    }
+
+    function managerHarvest() external onlyManager {
+        _harvest(tx.origin);
+    }
+
+    // compounds earnings and charges performance fee
+    function _harvest(address callFeeRecipient) internal whenNotPaused {
+        if (IComptroller(comptroller).pendingComptrollerImplementation() == address(0)) {
+            uint256 beforeBal = availableWant();
+            IComptroller(comptroller).claimComp(address(this), markets);
+            uint256 outputBal = IERC20(output).balanceOf(address(this));
+            if (outputBal > 0) {
+                chargeFees(callFeeRecipient);
+                swapRewards();
+                uint256 wantHarvested = availableWant() - beforeBal;
+                deposit();
+
+                lastHarvest = block.timestamp;
+                emit StratHarvest(msg.sender, wantHarvested, balanceOf());
+            }
+        } else {
+            panic();
+        }
+    }
+
+    // performance fees
+    function chargeFees(address callFeeRecipient) internal {
+        IFeeConfig.FeeCategory memory fees = getFees();
+        uint256 toNative = IERC20(output).balanceOf(address(this)) * fees.total / DIVISOR;
+        IUniswapRouterETH(unirouter).swapExactTokensForTokens(
+            toNative, 0, outputToNativeRoute, address(this), block.timestamp
+        );
+
+        uint256 nativeBal = IERC20(native).balanceOf(address(this));
+
+        uint256 callFeeAmount = nativeBal * fees.call / DIVISOR;
+        IERC20(native).safeTransfer(callFeeRecipient, callFeeAmount);
+
+        uint256 beefyFeeAmount = nativeBal * fees.beefy / DIVISOR;
+        IERC20(native).safeTransfer(beefyFeeRecipient, beefyFeeAmount);
+
+        uint256 strategistFeeAmount = nativeBal * fees.strategist / DIVISOR;
+        IERC20(native).safeTransfer(strategist, strategistFeeAmount);
+
+        emit ChargedFees(callFeeAmount, beefyFeeAmount, strategistFeeAmount);
+    }
+
+    // swap rewards to {want}
+    function swapRewards() internal {
+        uint256 outputBal = IERC20(output).balanceOf(address(this));
+        IUniswapRouterETH(unirouter).swapExactTokensForTokens(outputBal, 0, outputToWantRoute, address(this), block.timestamp);
+    }
+
+    /**
+     * @dev Withdraws funds and sends them back to the vault. It deleverages from venus first,
+     * and then deposits again after the withdraw to make sure it mantains the desired ratio.
+     * @param _amount How much {want} to withdraw.
+     */
+    function withdraw(uint256 _amount) external {
+        require(msg.sender == vault, "!vault");
+        require(balanceOfWant() >= _amount, "Want Balance Less than Requested");
+
+        uint256 wantBal = availableWant();
+
+        if (wantBal < _amount) {
+            _deleverage();
```

Comment:
Weso's comment placed the `require(balanceOfWant() >= _amount ...` statement inside this if() block.  Perhaps the code sample / bug bounty were from the SuppyOnly version of the strat, which doesn't have the _deleverage() logic broken out.  Are we better off putting it in _deleverage() where the .redeemUnderlying() call is actually made?